pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -675,3 +682,44 @@ type DigitalOceanSDConfig struct {
 	// +optional
 	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
 }
+
+// HetznerSDConfig allow retrieving scrape targets from Hetzner Cloud API and Robot API.
+// This service discovery uses the public IPv4 address by default, but that can be changed with relabeling
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#hetzner_sd_config
+// +k8s:openapi-gen=true
+type HetznerSDConfig struct {
+	// The Hetzner role of entities that should be discovered.
+	// One of robot or hcloud.
+	// +required
```

Comment:
IMHO I dont think we don't need new type for role here since its only two types so lets move the enum validation here directly
`// +kubebuilder:validation:Enum=hcloud;Hcloud;robot;Robot`